src/Modules/OrchardCore.Commerce/Controllers/PaymentController.cs

```diff
@@ -150,6 +157,59 @@ public async Task<IActionResult> Validate()
         }
     }
 
+    [Route("PaymentRequest/{orderId}")]
+    public async Task<IActionResult> PaymentRequest(string orderId)
+    {
+        if (await _contentManager.GetAsync(orderId) is not { } order) return NotFound();
+
+        // Users should only see their own Orders.
+        if (order.Author != User.Identity.Name)
+        {
+            return NotFound();
+        }
+
+        var orderPart = order.As<OrderPart>();
+
+        // If status is not Pending or there are no line items, there is nothing to be done.
+        if (!string.Equals(orderPart.Status.Text, OrderStatuses.Pending, StringComparison.OrdinalIgnoreCase) ||
+            !orderPart.LineItems.Any())
+        {
+            return NotFound();
+        }
+
+        var currency = orderPart.LineItems[0].LinePrice.Currency;
+        var singleCurrencyTotal = new Amount(0, currency);
+        foreach (var item in orderPart.LineItems)
+        {
+            singleCurrencyTotal += item.LinePrice;
+        }
```

Comment:
This can be simplified by passing `orderPart.LineItems.Sum()` to the constructor.

Reply:
It seems the method can be entirely replaced with a `Sum()` call: `orderTotals = orderPart.LineItems.Select(item => item.LinePrice).Sum();`

Reply:
Yeah, but you still pass an `Amount` instance to `CreatePaymentIntentAsync` so you might as well stick with creating it here.

Reply:
Whoops, I was looking at the wrong thing. `StripePaymentService`'s `CalculateOrderTotals()` is what I was referring to as something that can be removed. I see what you mean for this one, will update.